tests/integration/helpers/cluster.py

```diff
@@ -2480,7 +2480,20 @@ def start_rabbitmq_app(self, timeout=120):
         run_rabbitmqctl(
             self.rabbitmq_docker_id, self.rabbitmq_cookie, "start_app", timeout
         )
-        self.wait_rabbitmq_to_start()
+        self.wait_rabbitmq_to_start(timeout)
+
+    @contextmanager
+    def pause_rabbitmq(self, monitor=None, timeout=120):
+        if monitor is not None:
+            monitor.stop()
+        self.stop_rabbitmq_app(timeout)
+
+        try:
+            yield
+        finally:
+            self.start_rabbitmq_app(timeout)
+            if monitor is not None:
```

Comment:
Aren't you missing here a `rabbitmq_cluster.wait_rabbitmq_to_start()`?
Otherwise, once this finishes there's no guarantee for the next lines to be executed after the contextmanager that RabbitMQ is up

Reply:
`self.wait_rabbitmq_to_start` is already called from the `start_rabbitmq_app`

Reply:
Right, then I added that unnecessarily in tests/integration/test_storage_rabbitmq/test_failed_connection.py:136

Nice catch